### ISSUE TITLE
Time interval for code fetch from git repos

### ISSUE THREAD

**Issue by rohanvakharia** [sourcegraph/issues#53]
- Issue type: Question/Feature Request
- Sourcegraph version: 2.6.8
- OS Version: Ubuntu 16.04
- Docker version: 18.03.0-ce
```
Client:
 Version:       18.03.0-ce
 API version:   1.37
 Go version:    go1.9.4
 Git commit:    0520e24
 Built: Wed Mar 21 23:10:01 2018
 OS/Arch:       linux/amd64
 Experimental:  false
 Orchestrator:  swarm

Server:
 Engine:
  Version:      18.03.0-ce
  API version:  1.37 (minimum version 1.12)
  Go version:   go1.9.4
  Git commit:   0520e24
  Built:        Wed Mar 21 23:08:31 2018
  OS/Arch:      linux/amd64
  Experimental: false

```
Steps to reproduce:

I found that SourceGraph is too frequently fetching the gitlab(might be for other repos to but I am using gitlab) code. Which causes high load on gitlab instance.

I would like to know that is there any way I can control the interval?

If No, I would like you to consider this as feature request. So that It can be defined as time interval something like Hourly, Daily or specifying the minutes in the setting section.

**Comment by sqs** [comment]
Hi there, this is indeed configurable. See https://about.sourcegraph.com/docs/server/config/settings/#repolistupdateinterval-integer. 

Try setting it to 30 (minutes).

We are evaluating what other configurability is needed here, so let us know if this is sufficient or if you want other options (webhooks, per-repo intervals, etc.). Thanks!

cc @nicksnyder

**Comment by rohanvakharia** [comment]
@sqs Thanks for your quick reply. I would surely let you know.

**Comment by nicksnyder** [comment]
>  found cloning over ssh to significantly impact performance also, so have it cloning over http

@adam-moss Interesting, roughly how much slower did you find ssh to be?

**Comment by adam-moss** [comment]
@nicksnyder it was several orders of magnitude.  Doing normal dev activity via ssh we have no issues (that's what we use), but when I configured sourcegraph to use ssh also it also it was putting at ~60+s delay on simple git pushes.

My initial findings suggested queuing in the worker process was causing it it but I couldn't experiment too much with it as it was upsetting the dev teams 🤐 

Mind you that was early in the v2 series - a lot of the recent changes around timeouts and stuff are related to feedback I've been pushing to Dan.  He has some nagios graphs showing load on the gitlab tin concerned.

**Comment by nicksnyder** [comment]
> queuing in the worker process 

GitLab's worker process? I might file an issue against GitLab to see if this is a known performance consideration.

Thanks for the feedback, this will be useful information to share if other customers experience performance problems with GitLab.

**Comment by adam-moss** [comment]
Yep, I believe it was getting overloaded with the demand from sourcegraph.  Again though this was early in the v2 series when you couldn't adjust concurrency etc.

**Comment by nicksnyder** [comment]
We have a fix in master that makes sure our `git fetch`s are correctly throttled according to `gitMaxConcurrentClones`. Previously only the clones were throttled.

See also: https://github.com/sourcegraph/issues/issues/70#issuecomment-386646196

**Comment by nicksnyder** [comment]
This should be fixed in 2.8. Please reopen this issue, or open a new issue, if you continue to experience any problems.
